From 97d0afa11b44d295e49c67b978f786f7870f09e3 Mon Sep 17 00:00:00 2001
From: Jaroslav Mracek <jmracek@redhat.com>
Date: Fri, 11 Oct 2019 15:16:34 +0200
Subject: [PATCH] Use POOL_FLAG_WHATPROVIDESWITHDISABLED

It resolves the problem with --best and search in provides.
The patch requires libsolv-0.7.7.

https://bugzilla.redhat.com/show_bug.cgi?id=1737469

Requires for 100% functionality (build, run-time):
https://github.com/openSUSE/libsolv/commit/954862319e32e97b20afd0f598cd461ece320e4f
https://github.com/openSUSE/libsolv/commit/bfdc347fdfc350a9b290543fac905cb21f9a71a8

Tests:
https://github.com/openSUSE/libsolv/commit/7477a28ca57c9fd2df73e24a36d2a0f0734662ef
---
 libdnf.spec         | 2 +-
 libdnf/dnf-sack.cpp | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/libdnf.spec b/libdnf.spec
index 42205d0..a2bf2a2 100644
--- a/libdnf.spec
+++ b/libdnf.spec
@@ -1,4 +1,4 @@
-%global libsolv_version 0.7.4-1
+%global libsolv_version 0.7.7
 %global libmodulemd_version 1.6.1
 %global librepo_version 1.10.0
 %global dnf_conflict 4.2.11
diff --git a/libdnf/dnf-sack.cpp b/libdnf/dnf-sack.cpp
index 8848aab..57839a3 100644
--- a/libdnf/dnf-sack.cpp
+++ b/libdnf/dnf-sack.cpp
@@ -183,6 +183,7 @@ dnf_sack_init(DnfSack *sack)
 {
     DnfSackPrivate *priv = GET_PRIVATE(sack);
     priv->pool = pool_create();
+    pool_set_flag(priv->pool, POOL_FLAG_WHATPROVIDESWITHDISABLED, 1);
     priv->running_kernel_id = -1;
     priv->running_kernel_fn = running_kernel;
     priv->considered_uptodate = TRUE;
--
libgit2 0.28.2

